contracts/StrategyRegistry.vy

```diff
@@ -0,0 +1,70 @@
+# @version 0.2.12
+
+governance: public(address)
+pendingGovernance: public(address)
+
+event NewGovernance:
+    governance: address
+
+interface Strategy:
+    def name() -> String[64]: view
+
+struct StrategyInfo:
+    name: String[64]
+    numReleases: uint256 
+    versions: address[20]
+
+strategiesReleases: public(HashMap[bytes32, StrategyInfo])
+
+@external
+def __init__():
+    self.governance = msg.sender
+
+@external
+def setGovernance(governance: address):
+    """
+    @notice Starts the 1st phase of the governance transfer.
+    @dev Throws if the caller is not current governance.
+    @param governance The next governance address
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    self.pendingGovernance = governance
+
+
+@external
+def acceptGovernance():
+    """
+    @notice Completes the 2nd phase of the governance transfer.
+    @dev
+        Throws if the caller is not the pending caller.
+        Emits a `NewGovernance` event.
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    self.governance = msg.sender
+    log NewGovernance(msg.sender)
+
+@external
+def addNewRelease(strategy :address, name: String[64] = ""):
+    """
+    @notice add a new version of a strategy
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    strategyName: String[64] = name 
+    if name == "":
+        strategyName = Strategy(strategy).name()
+    strategyHash: bytes32 = sha256(strategyName)
+    if self.strategiesReleases[strategyHash].numReleases == 0:
+	    self.strategiesReleases[strategyHash] = StrategyInfo({
+		    name: strategyName,
+		    numReleases: 0,
+		    versions: [ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS]
+	    })
+    numReleases: uint256 = self.strategiesReleases[strategyHash].numReleases
+    self.strategiesReleases[strategyHash].numReleases = numReleases + 1
+    self.strategiesReleases[strategyHash].versions[numReleases] = strategy
+
```

Comment:
Not sure how we are going to handle the cloning inside here since the parameters will be different.
Have you thought of that issue?

Reply:
I don't think that is necessary. Find the strategy latest version and then use the clone function from the strategy

Reply:
well, if we are doing a registry I would like the registry to have the cloning logic.
Perhaps a factory contract? I am trying to trim as much fat as possible from strategies.

Reply:
I think the complicated part is that each strategy has its own params for the cloning logic. Isn't it?

Reply:
yep, that's why I was thinking that perhaps the registry can map to a factory.
Something like: `registry.getFactory(likeStrategy)`

Reply:
We also might decode the params and use a `bytes data` as an input param in the `cloneStrategy` function. For example, using a struct we might

```MyCustomStruct memory mcs = abi.decode(data, (MyCustomStruct));````

Reply:
Just an idea: 
In case a strategy is clonable:
- We call the factory.cloneStrategy(...)
- Factory call the stategy.clone(data) function.
- The strategy should decode the data bytes.
- Clone the strategy.
- Initialize the cloned strategy with the data.
- Finally, the factory should register the new cloned strategy

Reply:
So here I want to have two different registry, one that takes care of what's the most up to date version of a strategy (this contract) and an other one that can take care of keeping a list of all strategies (an other contract).

So the cloneStrategy should only be called by governance and then we can have a cloneStrategyExperiment function that can be done by anyone.